### ISSUE TITLE
AttributeError: 'TraitedSpec' object has no attribute 'AFNI_options'

### ISSUE THREAD

**Issue by nimzodisaster** [FCP-INDI/C-PAC#994]
CPAC 1.4.1
Error while processing pipeline config yaml:

Traceback (most recent call last):
  File "/home/jlee38/miniconda/envs/cpac/lib/python2.7/multiprocessing/process.py", line 258, in _bootstrap
    self.run()
  File "/home/jlee38/miniconda/envs/cpac/lib/python2.7/multiprocessing/process.py", line 114, in run
    self._target(*self._args, **self._kwargs)
  File "/home/jlee38/miniconda/envs/cpac/lib/python2.7/site-packages/CPAC/pipeline/cpac_pipeline.py", line 426, in prep_workflow
    anat_preproc.inputs.AFNI_options.set(
AttributeError: 'TraitedSpec' object has no attribute 'AFNI_options'

Note: I had to manually create my pipeline config file because the gui erred while saving it.
pipeline txt file
[REVISIONpipeline_config_proc_pt25_ICA_AROMA.txt](https://github.com/FCP-INDI/C-PAC/files/3180278/REVISIONpipeline_config_proc_pt25_ICA_AROMA.txt)

**Comment by nimzodisaster** [comment]
My data is already skull stripped. 
It does not appear to like:
# Disables skull-stripping on the anatomical inputs if they are already skull-stripped outside of C-PAC. Set this to On if your input images are already skull-stripped.
already_skullstripped :  [1]
# Choice of using AFNI or FSL-BET to perform SkullStripping
skullstrip_option :  AFNI

**Comment by nimzodisaster** [comment]
I may have solved this by putting an if statement before the anat_preproc.inputs.AFNI_options.set call. I think preproc doesn't return with AFNI options when already_skullstripped == 1 
This seemed to fix it:

```
if already_skullstripped == 0:
    anat_preproc.inputs.AFNI_options.set(
```

I'm not too git savvy so I won't submit a patch, but issue will likely affect other users so should be fixed.

**Comment by anibalsolon** [comment]
Thank you @nimzodisaster ! We will fix this, thanks for digging up the code :)

**Comment by nimzodisaster** [comment]
Perhaps that didn't help after all:
Pipeline continued but then crashed at:
```
190515-04:02:43,851 nipype.workflow ERROR:
	 Node Anatomical_2mm_flirt.a0.b0 failed to run on host rsmachine.
190515-04:02:43,856 nipype.workflow ERROR:
	 Saving crash info to /home/jlee38/EXECUTE/crsh/crash-20190515-040243-jlee38-Anatomical_2mm_flirt.a0.b0-0891650c-4707-496f-a4c9-9118d00a0d50.pklz
Traceback (most recent call last):
  File "/home/jlee38/miniconda/envs/cpac/lib/python2.7/site-packages/nipype/pipeline/plugins/linear.py", line 44, in run
    node.run(updatehash=updatehash)
  File "/home/jlee38/miniconda/envs/cpac/lib/python2.7/site-packages/nipype/pipeline/engine/nodes.py", line 471, in run
    result = self._run_interface(execute=True)
  File "/home/jlee38/miniconda/envs/cpac/lib/python2.7/site-packages/nipype/pipeline/engine/nodes.py", line 555, in _run_interface
    return self._run_command(execute)
  File "/home/jlee38/miniconda/envs/cpac/lib/python2.7/site-packages/nipype/pipeline/engine/nodes.py", line 623, in _run_command
    cmd = self._interface.cmdline
  File "/home/jlee38/miniconda/envs/cpac/lib/python2.7/site-packages/nipype/interfaces/base/core.py", line 948, in cmdline
    self._check_mandatory_inputs()
  File "/home/jlee38/miniconda/envs/cpac/lib/python2.7/site-packages/nipype/interfaces/base/core.py", line 389, in _check_mandatory_inputs
    raise ValueError(msg)
ValueError: FLIRT requires a value for input 'in_file'. For a list of required inputs, see FLIRT.help()
```

may have just pushed the issue down further

**Comment by nimzodisaster** [comment]
the pklz
```
Node: resting_preproc_112004-100_ses-1.nuisance_0_0.Anatomical_2mm_flirt
Working directory: /home/jlee38/EXECUTE/wrk/resting_preproc_112004-100_ses-1/nuisance_0_0/_scan_func-1/_selector_CSF-2mmE-M_M-SDB_BP-B0.008-T0.08_C-K-0+0-FD-P0.25/Anatomical_2mm_flirt


Node inputs:

angle_rep = <undefined>
apply_isoxfm = 2.0
apply_xfm = <undefined>
args = <undefined>
bbrslope = <undefined>
bbrtype = <undefined>
bgvalue = <undefined>
bins = <undefined>
coarse_search = <undefined>
cost = <undefined>
cost_func = <undefined>
datatype = <undefined>
display_init = <undefined>
dof = <undefined>
echospacing = <undefined>
environ = {u'FSLOUTPUTTYPE': 'NIFTI_GZ'}
fieldmap = <undefined>
fieldmapmask = <undefined>
fine_search = <undefined>
force_scaling = <undefined>
in_file = <undefined>
in_matrix_file = <undefined>
in_weight = <undefined>
interp = <undefined>
min_sampling = <undefined>
no_clamp = <undefined>
no_resample = <undefined>
no_resample_blur = <undefined>
no_search = <undefined>
out_file = <undefined>
out_log = <undefined>
out_matrix_file = <undefined>
output_type = NIFTI_GZ
padding_size = <undefined>
pedir = <undefined>
ref_weight = <undefined>
reference = <undefined>
rigid2D = <undefined>
save_log = <undefined>
schedule = <undefined>
searchr_x = <undefined>
searchr_y = <undefined>
searchr_z = <undefined>
sinc_width = <undefined>
sinc_window = <undefined>
uses_qform = <undefined>
verbose = <undefined>
wm_seg = <undefined>
wmcoords = <undefined>
wmnorms = <undefined>



Traceback: 
Traceback (most recent call last):
  File "/home/jlee38/miniconda/envs/cpac/lib/python2.7/site-packages/nipype/pipeline/plugins/linear.py", line 44, in run
    node.run(updatehash=updatehash)
  File "/home/jlee38/miniconda/envs/cpac/lib/python2.7/site-packages/nipype/pipeline/engine/nodes.py", line 471, in run
    result = self._run_interface(execute=True)
  File "/home/jlee38/miniconda/envs/cpac/lib/python2.7/site-packages/nipype/pipeline/engine/nodes.py", line 555, in _run_interface
    return self._run_command(execute)
  File "/home/jlee38/miniconda/envs/cpac/lib/python2.7/site-packages/nipype/pipeline/engine/nodes.py", line 623, in _run_command
    cmd = self._interface.cmdline
  File "/home/jlee38/miniconda/envs/cpac/lib/python2.7/site-packages/nipype/interfaces/base/core.py", line 948, in cmdline
    self._check_mandatory_inputs()
  File "/home/jlee38/miniconda/envs/cpac/lib/python2.7/site-packages/nipype/interfaces/base/core.py", line 389, in _check_mandatory_inputs
    raise ValueError(msg)
ValueError: FLIRT requires a value for input 'in_file'. For a list of required inputs, see FLIRT.help()

```

**Comment by nimzodisaster** [comment]
Here are the pklz crash files for one subject.
[crsh.zip](https://github.com/FCP-INDI/C-PAC/files/3184329/crsh.zip)

here is the content of the first occurring crash file:
```
File: /home/jlee38/treedrive/users/JKL/RS/EXECUTE/crsh/crash-20190515-040243-jlee38-Anatomical_2mm_flirt.a0.b0-0891650c-4707-496f-a4c9-9118d00a0d50.pklz
Node: resting_preproc_112004-100_ses-1.nuisance_0_0.Anatomical_2mm_flirt
Working directory: /home/jlee38/EXECUTE/wrk/resting_preproc_112004-100_ses-1/nuisance_0_0/_scan_func-1/_selector_CSF-2mmE-M_M-SDB_BP-B0.008-T0.08_C-K-0+0-FD-P0.25/Anatomical_2mm_flirt


Node inputs:

angle_rep = <undefined>
apply_isoxfm = 2.0
apply_xfm = <undefined>
args = <undefined>
bbrslope = <undefined>
bbrtype = <undefined>
bgvalue = <undefined>
bins = <undefined>
coarse_search = <undefined>
cost = <undefined>
cost_func = <undefined>
datatype = <undefined>
display_init = <undefined>
dof = <undefined>
echospacing = <undefined>
environ = {u'FSLOUTPUTTYPE': 'NIFTI_GZ'}
fieldmap = <undefined>
fieldmapmask = <undefined>
fine_search = <undefined>
force_scaling = <undefined>
in_file = <undefined>
in_matrix_file = <undefined>
in_weight = <undefined>
interp = <undefined>
min_sampling = <undefined>
no_clamp = <undefined>
no_resample = <undefined>
no_resample_blur = <undefined>
no_search = <undefined>
out_file = <undefined>
out_log = <undefined>
out_matrix_file = <undefined>
output_type = NIFTI_GZ
padding_size = <undefined>
pedir = <undefined>
ref_weight = <undefined>
reference = <undefined>
rigid2D = <undefined>
save_log = <undefined>
schedule = <undefined>
searchr_x = <undefined>
searchr_y = <undefined>
searchr_z = <undefined>
sinc_width = <undefined>
sinc_window = <undefined>
uses_qform = <undefined>
verbose = <undefined>
wm_seg = <undefined>
wmcoords = <undefined>
wmnorms = <undefined>



Traceback: 
Traceback (most recent call last):
  File "/home/jlee38/miniconda/envs/cpac/lib/python2.7/site-packages/nipype/pipeline/plugins/linear.py", line 44, in run
    node.run(updatehash=updatehash)
  File "/home/jlee38/miniconda/envs/cpac/lib/python2.7/site-packages/nipype/pipeline/engine/nodes.py", line 471, in run
    result = self._run_interface(execute=True)
  File "/home/jlee38/miniconda/envs/cpac/lib/python2.7/site-packages/nipype/pipeline/engine/nodes.py", line 555, in _run_interface
    return self._run_command(execute)
  File "/home/jlee38/miniconda/envs/cpac/lib/python2.7/site-packages/nipype/pipeline/engine/nodes.py", line 623, in _run_command
    cmd = self._interface.cmdline
  File "/home/jlee38/miniconda/envs/cpac/lib/python2.7/site-packages/nipype/interfaces/base/core.py", line 948, in cmdline
    self._check_mandatory_inputs()
  File "/home/jlee38/miniconda/envs/cpac/lib/python2.7/site-packages/nipype/interfaces/base/core.py", line 389, in _check_mandatory_inputs
    raise ValueError(msg)
ValueError: FLIRT requires a value for input 'in_file'. For a list of required inputs, see FLIRT.help()



(cpac)jlee38@:~/.../crsh$ nipypecli crash crash-20190515-132741-jlee38-resample_u-69b501c6-0069-4d26-8ccb-e0f18f64e84f.pklz
190515-13:29:52,524 nipype.utils ERROR:
	 Your Nipype version is: 1.1.9
190515-13:29:52,524 nipype.utils ERROR:
	 Nipype version of the pkl is: 1.1.2
Traceback (most recent call last):
  File "/usr/bin/nipypecli", line 9, in <module>
    load_entry_point('nipype==1.1.9', 'console_scripts', 'nipypecli')()
  File "/usr/lib/python2.7/dist-packages/click/core.py", line 716, in __call__
    return self.main(*args, **kwargs)
  File "/usr/lib/python2.7/dist-packages/click/core.py", line 696, in main
    rv = self.invoke(ctx)
  File "/usr/lib/python2.7/dist-packages/click/core.py", line 1060, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/lib/python2.7/dist-packages/click/core.py", line 889, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/lib/python2.7/dist-packages/click/core.py", line 534, in invoke
    return callback(*args, **kwargs)
  File "/usr/lib/python2.7/dist-packages/nipype/scripts/cli.py", line 94, in crash
    display_crash_file(crashfile, rerun, debug, dir)
  File "/usr/lib/python2.7/dist-packages/nipype/scripts/crash_files.py", line 51, in display_crash_file
    crash_data = loadcrash(crashfile)
  File "/usr/lib/python2.7/dist-packages/nipype/utils/filemanip.py", line 632, in loadcrash
    return loadpkl(infile, versioning=True)
  File "/usr/lib/python2.7/dist-packages/nipype/utils/filemanip.py", line 683, in loadpkl
    raise e
ImportError: No module named function
```

**Comment by nimzodisaster** [comment]
The first pklz crash file contents:
```
(cpac)jlee38@rsmachine:~/EXECUTE/crsh$ nipypecli crash crash-20190515-132741-jlee38-resample_u-69b501c6-0069-4d26-8ccb-e0f18f64e84f.pklz


File: /home/jlee38/EXECUTE/crsh/crash-20190515-132741-jlee38-resample_u-69b501c6-0069-4d26-8ccb-e0f18f64e84f.pklz
Node: resting_preproc_101939-200_1_ses-1.qc_snr_0.montage_snr.resample_u
Working directory: /home/jlee38/EXECUTE/wrk/resting_preproc_101939-200_1_ses-1/qc_snr_0/montage_snr/resample_u


Node inputs:

file_ = <undefined>
function_str = CPAC.qc.utils.resample_1mm



Traceback: 
Traceback (most recent call last):
  File "/home/jlee38/miniconda/envs/cpac/lib/python2.7/site-packages/nipype/pipeline/plugins/linear.py", line 44, in run
    node.run(updatehash=updatehash)
  File "/home/jlee38/miniconda/envs/cpac/lib/python2.7/site-packages/nipype/pipeline/engine/nodes.py", line 471, in run
    result = self._run_interface(execute=True)
  File "/home/jlee38/miniconda/envs/cpac/lib/python2.7/site-packages/nipype/pipeline/engine/nodes.py", line 555, in _run_interface
    return self._run_command(execute)
  File "/home/jlee38/miniconda/envs/cpac/lib/python2.7/site-packages/nipype/pipeline/engine/nodes.py", line 635, in _run_command
    result = self._interface.run(cwd=outdir)
  File "/home/jlee38/miniconda/envs/cpac/lib/python2.7/site-packages/nipype/interfaces/base/core.py", line 521, in run
    runtime = self._run_interface(runtime)
  File "/home/jlee38/miniconda/envs/cpac/lib/python2.7/site-packages/CPAC/utils/function.py", line 158, in _run_interface
    out = function_handle(**args)
TypeError: resample_1mm() takes exactly 1 argument (0 given)
```

**Comment by anibalsolon** [comment]
Yes! I could reproduce your errors and already fixed it. It will be released in the next C-PAC version.

**Comment by nimzodisaster** [comment]
Great!

**Comment by nimzodisaster** [comment]
I was hoping to take your fix and run it on 1.4.1, but now it complains of missing module lesion_preproc.
Is this a new feature of the next upcoming release? If I download  lesion_preproc.py am I likely to run into a compatibility issue?
Thanks

Josh

**Comment by nimzodisaster** [comment]
It seems a lot of the commit includes items that might not have strictly been about the bug I submitted, but I am unsure. Did the fix require al those changes? Is there a minimal fix I might be able to apply to 1.4.1 so I can get going on my data analysis? Sorry to pick your brain on this.

**Comment by nimzodisaster** [comment]
For those that are stuck on 1.4.1 with this issue and can't wait for the next version of CPAC, attached are my own patches that gets past the issue. The anat_preproc.py by anibalsolon and my edits for cpac_pipeline.py

[anat_preproc.py.txt](https://github.com/FCP-INDI/C-PAC/files/3188601/anat_preproc.py.txt)
[cpac_pipeline.py.txt](https://github.com/FCP-INDI/C-PAC/files/3188602/cpac_pipeline.py.txt)
